homeassistant/components/freedompro/__init__.py

```diff
@@ -0,0 +1,121 @@
+"""Support for freedompro."""
+import asyncio
+from datetime import timedelta
+import logging
+
+from pyfreedompro import get_list, get_states
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_API_KEY
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import aiohttp_client
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import COORDINATOR, DOMAIN, UNDO_UPDATE_LISTENER
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [
+    "light",
+]
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
```

Comment:
Move `hass.data.setdefault(DOMAIN, {})` to `async_setup_entry` so that `async_setup` can be removed.
A rebase is required.